### PR TITLE
Added default value

### PR DESCRIPTION
**Pull request by loonydev**
Hi guys,
Close https://github.com/47degrees/github4s/issues/549. 
Btw, some `Option` field already has `None` values. So It was mess. Some add `Option` only, some `Option = None`.

Also close https://github.com/47degrees/github4s/issues/526 by added
```scala
final case class Repository(
    ...
    permissions: Option[RepoPermissions] = None
)

final case class RepoPermissions(
    admin: Boolean,
    push: Boolean,
    pull: Boolean
)
```

### CODE REVIEW

File: github4s/src/main/scala/github4s/domain/Issue.scala [hunk 1]
```diff
@@ -30,31 +29,32 @@ final case class Issue(
...
 
 final case class IssuePullRequest(
-    url: Option[String],
-    html_url: Option[String],
-    diff_url: Option[String],
-    patch_url: Option[String]
+    url: String,
+    html_url: String,
+    diff_url: String,
+    patch_url: String
```
Comment (BenFradet) [inline github4s/src/main/scala/github4s/domain/Issue.scala:57]: it's highly likely that the options were there for a reason.

**Review by BenFradet — COMMENTED** [review]
LGTM, except for my comment :+1:

File: github4s/src/main/scala/github4s/domain/Issue.scala (hunk 1 above)
Comment (loonydev) [reply github4s/src/main/scala/github4s/domain/Issue.scala:57]: `IssuePullRequest` used in one place `Issue` where it's already `Option` field. I have check the [Issue docs](https://docs.github.com/en/rest/reference/issues) and don't see cases with some of this fields.

Well, I can change it back, but looks like overhead.

Comment (BenFradet) [reply github4s/src/main/scala/github4s/domain/Issue.scala:57]: Unfortunately, the documentation is not a reliable source of truth. we've been bit by issues like this in the past.

Comment (loonydev) [reply github4s/src/main/scala/github4s/domain/Issue.scala]: Sure, return option

**Comment by loonydev** [comment]
Hi @BenFradet, 
Do I need something else here?

**Review by BenFradet — APPROVED** [review]
Nope, thanks :+1:

**Pull request merged**